storagemarket/provider_loop.go

```diff
@@ -44,7 +46,7 @@ func (p *Provider) loop() {
 			go func() {
```

Comment:
If there's an error in any of the restart synchronization logic, we need to call `p.cleanupDealHandler(restartReq.deal.DealUuid)`
We're also calling cleanup if the context times out trying to put the restartReq into `restartDealsChan`. I wonder if there's some way to keep all the cleanup logic in one place 🤔 

Maybe it would be simplest to remove `restartDealsChan` for now, and only add it once we have restart synchronization logic. Otherwise we're adding code complexity for something we may not actually need.